### PR TITLE
[Arm64] Transform AND-CMP to TST

### LINKED ISSUE #13790: [Arm64] JIT optimize `if ((len & 24) == 0) goto `
`if ((len & 24) == 0) goto MCPY03;` is generating

```
        and     w0, w24, #24
        cmp     w0, #0
        beq     G_M55607_IG12
```
This could be
```
        ands    w0, w24, #24
        beq     G_M55607_IG12
```
or
```
        and     w0, w24, #24
        cbz     w0, G_M55607_IG12
```

### PR DESCRIPTION
**Pull request by sdmaclea**
Fixes #13790 

@mikden @dotnet/arm64-contrib @dotnet/jit-contrib  PTAL

### CODE REVIEW

**Comment by sdmaclea** [comment]
@mikedn Only slight changes were required to get this to pass regressions.  

JITDisam of `System.Buffer:Memmove()` was affected as expected.

```
      tst     w24, #24
      beq     G_M55607_IG12
```

**Comment by sdmaclea** [comment]
@dotnet-bot test Windows_NT x64 full_opt ryujit CoreCLR Perf Tests Correctness

**Comment by sdmaclea** [comment]
@dotnet-bot help

**Comment by sdmaclea** [comment]
@dotnet-bot test Windows_NT x86 full_opt legacy_backend CoreCLR Perf Tests Correctness
@dotnet-bot test Windows_NT x86 full_opt ryujit CoreCLR Perf Tests Correctness

**Comment by jashook** [comment]
@dontet-bot test Windows_NT Arm64 Checked

**Comment by sdmaclea** [comment]
> Minor restructuring suggestion.

@CarolEidt I cannot figure out what your suggestion is

File: src/jit/lower.cpp [hunk 1]
```diff
@@ -2454,7 +2455,12 @@ void Lowering::LowerCompare(GenTree* cmp)
                 }
             }
         }
+#endif
+#ifdef _TARGET_XARCH_
         else if (op1->OperIs(GT_AND) && cmp->OperIs(GT_EQ, GT_NE))
+#else
+        if (op1->OperIs(GT_AND) && cmp->OperIs(GT_EQ, GT_NE))
+#endif
```
Comment (CarolEidt) [inline src/jit/lower.cpp]: I would suggest to restructure the above 6 lines as:
```
        else
#endif \\ _TARGET_XARCH_
             if (op1->OperIs(GT_AND) && cmp->OperIs(GT_EQ, GT_NE))
```

Comment (sdmaclea) [reply src/jit/lower.cpp]: @CarolEidt done

**Comment by jashook** [comment]
@dotnet-bot test Windows_NT Arm64 Checked

**Comment by sdmaclea** [comment]
@dotnet-bot test Windows_NT Arm64 Checked
test Tizen armel Cross Release Build
test Windows_NT x64 Release Priority 1 Build and Test

**Comment by sdmaclea** [comment]
All tests have passed including Windows_NT Arm64 Checked.  This can be merged.

**Pull request merged**